client/src/react/components/MainScore.js

```diff
@@ -3,14 +3,32 @@ import './MainScore.css';
 
 import PlayerScore from './PlayerScore';
 
-function MainScore() {
-  const players = ['Marco Goebel', 'Felix Breitenbach'];
+function MainScore({ match }) {
+  const { player1, player2 } = match;
+
+  const calculateResult = player => {
+    let result = 0;
+
+    match.sets.forEach(set => {
+      if (player === 1) {
+        if (set.p1 >= 11 && set.p1 > set.p2) {
+          result++;
+        }
+      } else {
+        if (set.p2 >= 11 && set.p2 > set.p1) {
+          result++;
+        }
+      }
+    });
+
+    return result;
```

Comment:
a more succinct and functional approach would be to filter the sets and then just count how many there are, like so:

`const setsWon = player => opponent => sets.filter(set => set[player] >= 11 && set[player] - 2 >= set[opponent]).length`
`const setsWonPlayer1 = setsWon("player1")("player2")`
`const setsWonPlayer2 = setsWon("player2")("player1")`

or even

`const setsWonPlayer1 = sets.filter(set => set.player1 >= 11 && set.player1 - 2 >= set.player2).length`
`const setsWonPlayer2 = sets.filter(set => set.player2 >= 11 && set.player2 - 2 >= set.player1).length`

I probably like the first one better because it seems to describe the logic more clearly, what do you think?

Reply:
With the second solution we prevent that the programmer makes a typing error. It think that would the better solution.

Reply:
I moved the method into a seperate file and added a test

Reply:
I split the setsWon function into to seperate functions which use the setsWon function